packages/page-staking/src/Nomination/index.tsx

```diff
@@ -0,0 +1,380 @@
+import BN from 'bn.js';
+import CreateModal from '@polkadot/app-accounts/Accounts/modals/Create';
+import { useApi, useOwnStashes, useToggle} from '@polkadot/react-hooks';
+import { useTranslation} from '@polkadot/app-accounts/translate';
+import { DeriveStakingOverview } from '@polkadot/api-derive/types';
+import { Available } from '@polkadot/react-query';
+import { AddressInfo, Button, InputBalance, TxButton /*, Spinner*/ } from '@polkadot/react-components';
+import TabsHeader from '@polkadot/app-staking/Nomination/TabsHeader';
+import StashesTable from '@polkadot/app-staking/Nomination/StahesTable';
+import { useBalanceClear, useFees, WholeFeesType } from '@polkadot/app-staking/Nomination/useBalance';
+import { Balance } from '@polkadot/types/interfaces/runtime';
+import Summary from '@polkadot/app-staking/Nomination/Summary';
+import { formatBalance } from '@polkadot/util';
+import EraToTime from './EraToTime';
+import useValidators from './useValidators';
+import AccountSelector from './AccountSelector';
+import ControllerAccountSelector from './ControllerAccountSelector';
+
+const steps = ['choose', 'create', 'bond', 'nominate'];
+const stepInitialState = ['', 'disabled', 'disabled', 'disabled'];
+
+interface Props {
+  className?: string;
+  isVisible: boolean;
+  stakingOverview?: DeriveStakingOverview;
+  next?: string[];
+}
+
+function Nomination ({ className, isVisible, stakingOverview, next }: Props): React.ReactElement<Props> {
+  const { api } = useApi();
+  const [currentStep, setCurrentStep] = useState<string>(steps[0]);
+  const [alreadyHaveStashes /*, setAlreadyHaveStashes*/ ] = useState<boolean>(false);
+  const [isNominated, setIsNominated] = useState<boolean>(false);
+  const [controllerAccountId, setControllerAccountId] = useState<string | null>(null);
+  const [senderId, setSenderId] = useState<string | null>(null);
+  const [stepsState, setStepsState] = useState<string[]>(stepInitialState);
+  const [controllerAlreadyBonded, setControllerAlreadyBonded] = useState<boolean>(false);
+  const [isCreateOpen, toggleCreate] = useToggle();
+  const [selectedValidators, setSelectedValidators] = useState<string[]>([]);
+  const { wholeFees, feesLoading } : WholeFeesType = useFees(controllerAccountId, senderId, selectedValidators);
+  const [transferableAmount, setTransferableAmount] = useState<BN>(new BN(1));
+  const [amountToBond, setAmountToBond] = useState<BN | undefined>();
+  const [amount, setAmount] = useState<BN | undefined | null>(null);
+  const controllerBalance: Balance | null = useBalanceClear(controllerAccountId);
+  const accountBalance: Balance | null = useBalanceClear(senderId);
+  const ownStashes = useOwnStashes();
+  const { validators, validatorsLoading } = useValidators();
+  const { t } = useTranslation();
+  const destination = 2; // 2 means controller account
+  const extrinsic = (amount && controllerAccountId)
+    ? api.tx.staking.bond(controllerAccountId, amount, destination)
+    : null;
```

Comment:
This gets done each an every render, it is expensive. You really don't want to follow this approach, rather effect. If at all possible, just the name & params to `TxButton`